From 156c25324fce53752ae1a0bddb20d0c153b63f7a Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Wed, 21 Jul 2021 00:18:18 +0200
Subject: [PATCH 3/4] ALSA: als300: Fix missing chip initialization

The recent code refactoring missed the initialization of the chip
variable as its allocation was moved to card->private_data.
Let's fix it.

Fixes: 21a9314cf93b ("ALSA: als300: Allocate resources with device-managed APIs")
Reported-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/s5hh7goocid.wl-tiwai@suse.de
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Link: https://git.kernel.org/tiwai/sound/c/f263a2c2d6c7c44e9256e6a24c9fec668f22ff8b
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 sound/pci/als300.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/pci/als300.c b/sound/pci/als300.c
index 9c94072572a5..b86565dcdbe4 100644
--- a/sound/pci/als300.c
+++ b/sound/pci/als300.c
@@ -702,6 +702,7 @@ static int snd_als300_probe(struct pci_dev *pci,
 				sizeof(*chip), &card);
 	if (err < 0)
 		return err;
+	chip = card->private_data;
 
 	chip_type = pci_id->driver_data;
 
-- 
2.32.0.264.g75ae10bc75

